snorkel/labeling/lf/nlp.py

```diff
@@ -0,0 +1,210 @@
+from typing import Any, Callable, List, Mapping, NamedTuple, Optional
+
+from snorkel.labeling.preprocess import BasePreprocessor
+from snorkel.labeling.preprocess.nlp import EN_CORE_WEB_SM, SpacyPreprocessor
+
+from .core import LabelingFunction
+
+
+class SpacyPreprocessorParameters(NamedTuple):
+    """Parameters needed to construct a SpacyPreprocessor.
+
+    See ``snorkel.labeling.preprocess.nlp.SpacyPreprocessor``.
+    """
+
+    text_field: str
+    doc_field: str
+    language: str
+    disable: Optional[List[str]]
+    preprocessors: List[BasePreprocessor]
+    memoize: bool
+
+
+class SpacyPreprocessorConfig(NamedTuple):
+    """Tuple of SpacyPreprocessor and the parameters used to construct it."""
+
+    nlp: SpacyPreprocessor
+    parameters: SpacyPreprocessorParameters
+
+
+class NLPLabelingFunction(LabelingFunction):
+    """Special labeling function type for SpaCy-based LFs.
+
+    This class is a special version of ``LabelingFunction``. It
+    has a ``SpacyPreprocessor`` integrated which shares a cache
+    with all other ``NLPLabelingFunction`` instances. This makes
+    it easy to define LFs that have a text input field and have
+    logic written over SpaCy ``Doc`` objects. Examples passed
+    into an ``NLPLabelingFunction`` will have a new field which
+    can be accessed which contains a SpaCy ``Doc``. By default,
+    this field is called ``doc``. For details of SpaCy ``Doc``
+    objects and a full attribute listing, see https://spacy.io/api/doc.
+
+    Simple ``NLPLabelingFunction``s can be defined via a
+    decorator. See ``nlp_labeling_function``.
```

Comment:
Could we have (a) a quick example here (just copy-paste the one from the PR description?) and (b) a partial listing of the top 5/10 spacy attributes (people will be lazy re: following the link)

Reply:
Or just copy from decorator version below?